src/main/java/hudson/remoting/forward/CopyThread.java

```diff
@@ -29,6 +42,7 @@ public void run() {
                 int len;
                 while ((len = in.read(buf)) > 0)
                     out.write(buf, 0, len);
+                throw new IllegalArgumentException();
```

Comment:
Needs some comments and maybe exception message around.
It should be likely IllegalStateException instead.